### PR TITLE
Add game states

### PR DESCRIPTION
**Pull request by thearst3rd**
- [x] Implement countdown state
- [x] Implement game-end state ~~and show scores~~ (not showing scores currently but those will show on the lobby screen)

### CODE REVIEW

File: resources/fonts/italic_large.tres [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[gd_resource type="FontVariation" format=3 uid="uid://bbq4hy4ot1oxr"]
+
+[resource]
+variation_transform = Transform2D(1, 0.2, 0, 1, 0, 0)
```
Comment (admoore0) [inline resources/fonts/italic_large.tres:4]: image

File: src/states/game.gd [hunk 2]
```diff
@@ -14,8 +23,11 @@ const ArrowPickup = preload("res://src/objects/arrow_pickup.tscn")
 @onready var pause_menu: Control = %PauseMenu
 @onready var scoreboard: Scoreboard = %Scoreboard
 @onready var arrows: Node = %Arrows
+@onready var match_timer: Label = %MatchTimer
+@onready var spawn_countdown: Label = %SpawnCountdown
 @onready var power_indicator: Control = %PowerIndicator
 @onready var quiver_display: Label = %QuiverDisplay
+@onready var animation_player: AnimationPlayer = %AnimationPlayer
```
Comment (admoore0) [inline src/states/game.gd:30]: What is this animation player for?

File: src/states/menus/lobby.gd [hunk 3]
```diff
@@ -95,17 +95,23 @@ func on_back_button_press() -> void:
...
 
 # Start the game
-@rpc("any_peer")
+@rpc("authority", "call_local")
 func start_game() -> void:
 	for player_id in chosen_colors.keys():
 		Multiplayer.player_info[player_id].color = COLORS[chosen_colors[player_id]]
 		Multiplayer.player_info[player_id].team_id = chosen_colors[player_id]
+	#server_name.text = "fake loading lol"
+	#get_tree().create_timer(3).timeout.connect(get_tree().change_scene_to_file.bind("res://src/states/game.tscn"))
```
Comment (admoore0) [inline src/states/menus/lobby.gd]: Remove? I address in my PR.

Comment (kdreese) [inline src/states/menus/lobby.gd]: Get rid pls

File: resources/fonts/italic_large.tres (hunk 1 above)
Comment (thearst3rd) [reply resources/fonts/italic_large.tres:4]: literally so true

File: src/states/game.gd (hunk 2 above)
Comment (thearst3rd) [reply src/states/game.gd:30]: For the 3 2 1 Go! countdown

File: src/states/menus/lobby.gd (hunk 3 above)
Comment (thearst3rd) [reply src/states/menus/lobby.gd]: donezo

Comment (thearst3rd) [reply src/states/menus/lobby.gd]: donezo

**Review by admoore0 — APPROVED** [review]

**Review by kdreese — APPROVED** [review]

**Pull request merged**